### PR TITLE
Fix codacy and coverage

### PR DESCRIPTION
**Pull request by jvanhoefer**
This PR basically tries to bring up the codacy rating & the coverage. I therefore

- [x] fixed some of the codacy issues
- [x] went over all the doc strings with the `pydocstyle` checker. Sorry for the many minor changes.
- [x] included the shell script tests into pytest in `test_CLI.py`, since this is more consistent and  the coverage should get these parts, while currently the CLI tests are overlooked (which is in large parts the reason for the bad coverage values...)

### CODE REVIEW

**Comment by jvanhoefer** [comment]
As discussed elsewhere: The open codacy issues are a bit ridiculous. I used `pydocstyle`, that legibly is used by codacy themselves, to tidy up all doc strings. And now codacy is complaining about exactly these changes. Therefore from my side I would leave things as they are. Ready for review + potentially merging :)

File: doc/examples/Finite_State_Projection/utils.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+"""Utility functions, that simulate and plot the FSP example."""
```
Comment (yannikschaelte) [inline doc/examples/Finite_State_Projection/utils.py:1]:
```suggestion
"""Utility functions which simulate and plot the FSP example."""
```

File: doc/examples/Finite_State_Projection/utils.py [hunk 2]
```diff
@@ -12,9 +13,7 @@ def plot_AMICI(sbml_dir: str,
                t: np.ndarray,
                r_max: int,
                p_max: int):
-    """
-    Compiles, simulates and plots the AMICI model for the FSP example.
-    """
+    """Compiles, simulates and plots the AMICI model for the FSP example."""
```
Comment (yannikschaelte) [inline doc/examples/Finite_State_Projection/utils.py:16]:
```suggestion
    """Compile, simulate and plot the AMICI model for the FSP example."""
```
active/imperative usually preferable

File: doc/examples/Sorensen/utils.py [hunk 3]
```diff
@@ -9,8 +10,9 @@
 
 
 def simulate_and_plot_sorensen(sbml_dir: str):
-    """Plots and simulates the Sorensen model.
-    
+    """
+    Plot and simulate the Sorensen model.
```
Comment (yannikschaelte) [inline doc/examples/Sorensen/utils.py:14]:
```suggestion
    """Plot and simulate the Sorensen model.
```

File: doc/examples/Sorensen/utils.py [hunk 4]
```diff
@@ -54,7 +56,8 @@ def simulate_and_plot_sorensen(sbml_dir: str):
 
 
 def compile_model(sbml_dir: str):
-    """Compiles the Sorensen model and specifies the observables.
+    """
+    Compiles the Sorensen model and specifies the observables.
```
Comment (yannikschaelte) [inline doc/examples/Sorensen/utils.py:60]:
```suggestion
    """Compile the Sorensen model and specify the observables.
```

File: tests/test_CLI.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    yaml_dir = os.path.join(path, 'test_yaml2sbml/ode_input1.yaml')
+    sbml_dir = os.path.join(path, 'test_sbml.xml')
+
+    script_runner.run('yaml2sbml', yaml_dir, sbml_dir)
+
+    os.remove(sbml_dir)
+
+
+def test_yaml2petab_cli(script_runner):
+    """Test the command line command yaml2petab."""
```
Comment (yannikschaelte) [inline tests/test_CLI.py:18]:
```suggestion
    """Test the command line command `yaml2petab`."""
```

File: tests/test_CLI.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                      output_dir,
+                      model_name,
+                      '-y test_yaml.yml')
+
+    # delete the generated files
+    shutil.rmtree(output_dir)
+
+
+def test_yaml2sbml_validate_cli(script_runner):
+    """Test the command line command yaml2sbml_validate"""
```
Comment (yannikschaelte) [inline tests/test_CLI.py:39]:
```suggestion
    """Test the command line command `yaml2sbml_validate`"""
```

File: tests/test_CLI.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+def test_yaml2sbml_cli(script_runner):
+    """Test the command line command `yaml2sbml`."""
+    path = os.path.dirname(os.path.abspath(__file__))
+
+    yaml_dir = os.path.join(path, 'test_yaml2sbml/ode_input1.yaml')
+    sbml_dir = os.path.join(path, 'test_sbml.xml')
+
+    script_runner.run('yaml2sbml', yaml_dir, sbml_dir)
+
+    os.remove(sbml_dir)
```
Comment (yannikschaelte) [inline tests/test_CLI.py:14]: kinda trivial, but maybe compare that the output is the same as one obtained via the python API?

File: tox.ini [hunk 8]
```diff
@@ -13,6 +13,7 @@ envlist =
     pyroma
     flake8
     unittests
+    pytest
```
Comment (yannikschaelte) [inline tox.ini:16]: I think this line should not be necessary. the "env"s specified here correspond to the `testenv:SOMETHING` environments.

**Review by yannikschaelte — APPROVED** [review]
Didn't know pytest-console-scripts, looks good!

File: doc/examples/Finite_State_Projection/utils.py (hunk 1 above)
Comment (dilpath) [reply doc/examples/Finite_State_Projection/utils.py:1]: Still should be `that` without comma (`which` with comma)

Comment (yannikschaelte) [reply doc/examples/Finite_State_Projection/utils.py:1]: true, you win :rofl:

File: yaml2sbml/yaml2sbml.py [hunk 9]
```diff
@@ -277,16 +241,12 @@ def _read_assignments_block(model: sbml.Model, assignment_list: list):
 
 def _create_assignment(model: sbml.Model, assignment_id: str, formula: str):
     """
-    Creates an  assignment rule, that assigns id to formula.
+    Create an assignment rule, that assigns <id> to <formula>.
```
Comment (dilpath) [inline yaml2sbml/yaml2sbml.py:244]:
```suggestion
    Create an assignment rule, that assigns `formula` to `assignment_id`.
```

File: yaml2sbml/yaml2sbml.py [hunk 10]
```diff
@@ -352,8 +301,9 @@ def _create_function(model: sbml.Model,
 
 def _read_odes_block(model: sbml.Model, odes_list: list):
     """
-    Reads and processes the odes block in the ODE yaml file.
-    In particular, it reads the odes and adds a species for the corresponding
+    Read and process the odes block in the YAML file.
```
Comment (dilpath) [inline yaml2sbml/yaml2sbml.py:304]:
```suggestion
    Read and process the `odes` block in the YAML file.
```

File: yaml2sbml/yaml2sbml.py [hunk 11]
```diff
@@ -352,8 +301,9 @@ def _create_function(model: sbml.Model,
...
+
+    In particular, read the odes and add a species for the corresponding
```
Comment (dilpath) [inline yaml2sbml/yaml2sbml.py:306]:
```suggestion
    In particular, read the ODEs and add a species for the corresponding
```

File: yaml2sbml/yaml2sbml.py [hunk 12]
```diff
@@ -376,7 +321,8 @@ def _read_odes_block(model: sbml.Model, odes_list: list):
 
 def _create_species(model: sbml.Model, species_id: str, initial_amount: str):
     """
-    Creates a species and adds it to the given SBML model.
+    Create a species and adds it to the SBML model.
```
Comment (dilpath) [inline yaml2sbml/yaml2sbml.py:324]:
```suggestion
    Create a species and add it to the SBML model.
```

File: yaml2sbml/yaml2sbml.py [hunk 13]
```diff
@@ -435,19 +374,15 @@ def _create_rate_rule(model: sbml.Model, species: str, formula: str):
 
 def _read_observables_block(model: sbml.Model, observable_list: list):
     """
-    Reads an processes the observables block in the ODE yaml file.
+    Read an process the observables block in the YAML file.
```
Comment (dilpath) [inline yaml2sbml/yaml2sbml.py:377]:
```suggestion
    Read and process the `observables` block in the YAML file.
```

File: yaml2sbml/yaml2sbml.py [hunk 14]
```diff
@@ -456,26 +391,23 @@ def _read_observables_block(model: sbml.Model, observable_list: list):
 
 def _read_conditions_block(model: sbml.Model, conditions_list: list):
     """
-    Reads an processes the conditions block in the ODE yaml file.
-    Since the conditions are not represented in the SBML, it only gives
+    Read an process the conditions block in the YAML file.
```
Comment (dilpath) [inline yaml2sbml/yaml2sbml.py:394]:
```suggestion
    Read and process the `conditions` block in the YAML file.
```

File: yaml2sbml/yaml2sbml.py [hunk 15]
```diff
@@ -456,26 +391,23 @@ def _read_observables_block(model: sbml.Model, observable_list: list):
...
-    Raises:
-
     """
     warnings.warn(
         'Conditions are not represented in the SBML and therefore only have '
         'an effect on the output when called via yaml2PEtab')
 
 
 def main():
+    """Command Line Interface."""
```
Comment (dilpath) [inline yaml2sbml/yaml2sbml.py:410]:
```suggestion
    """Command-line Interface."""
```

File: yaml2sbml/yaml_validation.py [hunk 16]
```diff
@@ -1,3 +1,4 @@
+"""Validator of the input yaml."""
```
Comment (dilpath) [inline yaml2sbml/yaml_validation.py:1]:
```suggestion
"""Validator of the input YAML."""
```

File: yaml2sbml/yaml_validation.py [hunk 17]
```diff
@@ -9,16 +10,13 @@
 
 def validate_yaml(yaml_file: str):
     """
-    Validates the syntax of the yaml file.
+    Validate the syntax of the yaml file.
```
Comment (dilpath) [inline yaml2sbml/yaml_validation.py:13]:
```suggestion
    Validate the syntax of the YAML file.
```

File: yaml2sbml/yaml_validation.py [hunk 18]
```diff
@@ -31,16 +29,13 @@ def validate_yaml(yaml_file: str):
 
 def _validate_yaml_from_dict(yaml_dict: dict):
     """
-    Validates the syntax of the yaml file, using a dict as input.
+    Validate the syntax of the yaml file, using a dict as input.
```
Comment (dilpath) [inline yaml2sbml/yaml_validation.py:32]:
```suggestion
    Validate the syntax of the YAML file, using a `dict` as input.
```

File: yaml2sbml/yaml_validation.py [hunk 19]
```diff
@@ -51,9 +46,7 @@ def _validate_yaml_from_dict(yaml_dict: dict):
 
 
 def main():
-    """
-    Function called by the CLI.
-    """
+    """Command Line Interface."""
```
Comment (dilpath) [inline yaml2sbml/yaml_validation.py:49]:
```suggestion
    """Command-line Interface."""
```

File: yaml2sbml/YamlModel.py [hunk 20]
```diff
@@ -28,7 +25,7 @@ def __init__(self):
     @staticmethod
     def load_from_yaml(yaml_dir):
         """
-        Creates a model instance from a yaml file.
+        Create a model instance from a yaml file.
```
Comment (dilpath) [inline yaml2sbml/YamlModel.py:28]:
```suggestion
        Create a model instance from a YAML file.
```

**Review by dilpath — APPROVED** [review]
:+1:

File: tests/test_CLI.py (hunk 7 above)
Comment (jvanhoefer) [reply tests/test_CLI.py:14]: good idea, included something like this.

**Pull request merged**